azure_arc_k8s_jumpstart/docs/aro_terraform.md

```diff
@@ -0,0 +1,232 @@
+# Overview
+
+The following README will guide you on how to use the Azure CLI to deploy an Azure Red Hat OpenShift 4 cluster connected it as an Azure Arc cluster resource.
+
+# Prerequisites
+
+* Create Azure Service Principal (SP) **(This can be done in Cloud Shell as well)**
+
+    To connect the EKS cluster to Azure Arc, Azure Service Principal is required and can be created by:
+
+
+    ```
+    az login
+
+    az ad sp create-for-rbac -n "http://AzureArcK8s" --role contributor
+    ``` 
+
+    Output should look like this:
+    ```
+    {
+       "appId": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+       "displayName": "AzureArcK8s",
+       "name": "http://AzureArcK8s",
+       "password": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+       "tenant": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX"
+    }
+    ```
+    **Note**: It is optional but highly recommended to scope the SP to a specific [Azure subscription and Resource Group](https://docs.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest) 
+
+* Enable subscription for the three providers: <br>
+   * Two for Azure Arc enabled Kubernetes
+   * One for RedHat OpenShift clusters
+
+  Registration is an asynchronous process, and registration may take approximately 10 minutes.
+
+  ```bash
+  az provider register --namespace Microsoft.Kubernetes
+  Registering is still on-going. You can monitor using 'az provider show -n Microsoft.Kubernetes'
+
+  az provider register --namespace Microsoft.KubernetesConfiguration
+  Registering is still on-going. You can monitor using 'az provider show -n Microsoft.KubernetesConfiguration'
+
+  az provider register -namespace Microsoft.RedHatOpenShift
+    Registering is still on-going. You can monitor using 'az provider show -n Microsoft.RedHatOpenShift
+  ```
+  You can monitor the registration process with the following commands:
+  ```bash
+  az provider show -n Microsoft.Kubernetes -o table
+ 
+  az provider show -n Microsoft.KubernetesConfiguration -o table
+
+  az provider show -n Microsoft.RedHatOpenShift -o table
+  ```
+
+* Install the ***connectedk8s***, ***k8sconfiguration*** and ***aro*** extension, which helps you deploy an Azure RedHat Openshift cluster and connect it to Arc Kubernetes cluster service.
+
+  ```bash
+  az extension add --name connectedk8s
+  az extension add --name k8sconfiguration
+  az extension add --name aro --index https://az.aroapp.io/stable
+  ```
+
+# Deployment
+There are two sets of resources that will be deployed, first is the Azure RedHat Openshift Container cluster. Second is the Azure Arc Kubernetes resource that will connect the ```aro``` cluster to Azure Arc.
+
+The deployment of all resources is going to be performed on Azure Cloud Shell.
```

Comment:
"on Azure Cloud Shell" > "in Azure Cloud Shell"

Reply:
done